R-package/cran-comments.md

```diff
@@ -1,9 +1,21 @@
 # CRAN Submission History
 
-## v3.0.1 - Submission 1 - (November 15, 202)
+## v3.1.0.1 - Submission 1 - (November 21, 2020)
```

Comment:
> I'll submit 3.1.0.1 to CRAN with an explanation about why we are submitting again so soon.

I think it can be `3.1.1` regular release for the whole project as we haven't agreed on #3210 yet (4-part versioning).

Reply:
I'm fine with doing a 3.1.1 release of the whole project, but that generates some extra work for you and @guolinke , so I thought it would be simpler to do an R-only version change.

adding a 4th version number just for the purpose of getting to CRAN is what we did in the last submission cycle (https://github.com/microsoft/LightGBM/pull/3338#issuecomment-698724866).

Let's let @guolinke be the tie-break (choose either 3.1.0.1 or 3.1.1 for the new CRAN submission). If it's 3.1.1, I can update this PR accodingly.

Reply:
> adding a 4th version number just for the purpose of getting to CRAN is what we did in the last submission cycle

Please note that CI cannot be run due to merge conflicts. You simply cannot generate artifact that should be uploaded to CRAN (you can do it only manually on your machine, but then what was the reason to add `r_artifacts.yml` workflow?..)

Reply:
> you can do it only manually on your machine, but then what was the reason to add r_artifacts.yml workflow?

So that checks that need to be run manually, like the Solaris ones, don't have to always be done by me, in case I'm unavailable. Anyone could generate the package on GitHub Actions with that comment, then upload it to https://builder.r-hub.io/.

I just looked at the list of recently-merged PRs and it doesn't seem like there are any new features, just small fixes, so I'm fine with making this 3.1.1. I'll update this branch to the latest `master` and change the version.

Reply:
we can update it to 3.1.1, but only update it cran.

Reply:
> we can update it to 3.1.1, but only submit it cran.

I guess it will be confusing for users. And we will have a bunch of issues with a content like "When `3.1.1` version will be released at PyPI?"

Reply:
I just check the commit log, there are only “maintenance” change, not any bug fixes or new features. Therefore, I don't think we should release a new version.

Reply:
besides, releasing a new version in such a short time seems we have an emergence bug fix.

Reply:
> I don't think we should release a new version.

Ok, then I think we should use 3.1.0.1 as the version number for CRAN. We have to increment the version number when we re-submit.

By the way @guolinke , has CRAN emailed you? I think they should be done running all their checks.

There are 12 checks done at  https://cran.r-project.org/web/checks/check_results_lightgbm.html

![image](https://user-images.githubusercontent.com/7608904/99924999-ef6c4700-2d01-11eb-9c93-06f156d79d18.png)

which is how many all other packages have (for example, `{jsonlite}`)

![image](https://user-images.githubusercontent.com/7608904/99925023-0317ad80-2d02-11eb-8594-e3b8f733f546.png)

But I agree with @StrikerRUS 's comment in https://github.com/microsoft/LightGBM/pull/3586#issuecomment-731668240 that we shouldn't submit again until CRAN asks.

Reply:
> I just check the commit log, there are only “maintenance” change, not any bug fixes or new features. 

Today we merged 2 bugfixes. And there can be more "fix" or "features" PRs if we wait some days for CRAN response.

![image](https://user-images.githubusercontent.com/25141164/99971479-42033d00-2dae-11eb-80ac-33b439433d0f.png)